cpp/locality/NeighborQuery.h

```diff
@@ -265,7 +265,20 @@ class NeighborQueryPerPointIterator : public NeighborPerPointIterator
         : NeighborPerPointIterator(query_point_idx), m_neighbor_query(neighbor_query),
           m_query_point(query_point), m_finished(false), m_r_max(r_max), m_r_min(r_min),
           m_exclude_ii(exclude_ii)
-    {}
+    {
+        if (r_max <= 0)
+    {
+        throw std::invalid_argument("RDF requires r_max to be positive.");
```

Comment:
This is the correct place to implement the validation logic for the `NeighborQuery` classes like `AABBQuery` and `LinkCell`. 👍  There are a couple changes needed.
1. Change the message:
```suggestion
        throw std::invalid_argument("Neighbor queries require r_max to be positive.");
```
2. Fix the indentation. You can run `clang-format` on the repository or just look at the rest of the code to learn the indentation style. See the developer documentation for info: https://freud.readthedocs.io/en/latest/reference/development/howtoadd.html